### PR TITLE
fix: do not mutate live when managed namespace enabled

### PR DESCRIPTION
**Pull request by leoluz**
While testing managed namespaces locally I noticed 2 issues:
- When enabled, managed namespace shouldn't mutate the live state directly.
- We can't set resource tracking without changing the diff logic to handle it correctly. More details in #11196

Signed-off-by: Leonardo Luz Almeida <leonardo_almeida@intuit.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: controller/sync_namespace.go
Comment (leoluz) [inline controller/sync_namespace.go:40]: Usually I don't do this but I left this intentionally committed so the person who works on #11196 has a partial implementation ready.

**Comment by leoluz** [comment]
@crenshaw-dev and other possible reviewers: Once approved, please do not merge. I need to update gitops-engine dependency in this PR first.

File: controller/sync_namespace.go
Comment (blakepettersson) [reply controller/sync_namespace.go:40]: Would it be sufficient to pass in an instance of `appStateManager` and to then do a pre-check with `appStateManager.isSelfReferencedObj`?  So for this to work as discussed in #11196, in order to return an error we'd do something like below? 

```suggestion
		selfReferencedObj := appStateManager.isSelfReferencedObj(liveNs, managedNs, appName, appLabelKey, trackingMethod)
		if !selfReferencedObj {
			return false, fmt.Errorf("insert error here")
		}

		err := resourceTracking.SetAppInstance(managedNs, appLabelKey, appName, "", trackingMethod)
		if err != nil {
			return false, fmt.Errorf("failed to set app instance tracking on the namespace %s: %s", managedNs.GetName(), err)
		}
```

Comment (leoluz) [reply controller/sync_namespace.go:40]: @blakepettersson Exactly! This would address this conflict validation logic that we discussed in #11196 . 
But I'm keeping this outside of this PR as I'm fixing another issue here.

File: test/e2e/app_management_ns_test.go
Comment (crenshaw-dev) [inline test/e2e/app_management_ns_test.go:1873]:
```suggestion
	t.Setenv("ARGOCD_GPG_ENABLED", "false")
```

**Review by crenshaw-dev — APPROVED** [review]

**Pull request merged**
